doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,101 @@
+# Operation sequences
+
+## Problem
+
+The current public API allows various kind of sequences of `kdbGet` and `kdbSet` calls.
+There is no real guideline of allowed and/or prohibited sequences, other than `kdbGet` for a certain parent key must be called before `kdbSet`:
+
+- initially after `kdbOpen`
+- after conflict errors in `kdbSet`
+
+While researching ways to do change tracking, we discovered that some sequences are currently legal but might be problematic.
+This issue not only effects change tracking, but may cause problems for other use cases too.
+
+Currently, plugins may (and in the case of notification plugins actually do) assume that a `kdbSet` operation for a certain parent key always directly follows the `kdbGet` operation for that same key. 
+However, this is not enforced in any way within Elektra, nor is it documented that it should.
+
+So in reality, as illustrated by the example, `kdbGet` and `kdbSet` of different parent keys might be interwoven.
+
+```c
+Key * keyA = ...;
+Key * keyB = ...;
+// assume that neither A below B nor B below A
+KeySet * a = ...;
+KeySet * b = ...;
+kdbGet(kdb, a, keyA); // (A)
+kdbGet(kdb, b, keyB); // (B)
+kdbSet(kdb, a, keyA); // (C)
+```
+
+- At point `(A)` the plugins internally store the KeySet for key `a`
+- At point `(B)` the plugins internally override the stored KeySet with the KeySet for key `b`
+- At point `(C)` plugins that do change-tracking will generate a wrong changeset for key `a` if they assume that the previous `kdbGet` operation was also for key `a`
```

Comment:
```suggestion
- At point `(A)` the plugins internally store the data below the parent `keyA`
- At point `(B)` the same thing happens for the data below the parent `keyB`.
  This **replaces** the data from `(A)`.
- At point `(C)` plugins that do change-tracking need to generate a changeset.
  The changeset will be calculated between the stored data and the new data in the KeySet `a`.
   Because the stored data is now from `(B)` and **not** from `(A)`, this changeset will be wrong.
   The calculation will wrongly assume everything in KeySet `b` was removed and everything in KeySet `a` is newly added.
```